src/components/Text/Text.tsx

```diff
@@ -1,20 +1,23 @@
 import React from 'react';
 import styled from '@emotion/styled';
 
-interface TextProps {
+import { themeHelper, sxPropHelper, StyleProps } from '../../utils/styled';
+
+interface TextProps extends StyleProps {
     children: React.ReactNode;
     level: 1 | 2 | 3 | 4 | 5;
-    as?: React.ElementType;
 }
 
-const Text = styled.span<TextProps>((props) => ({
-    fontSize: (props.theme as any).fontSizes[props.level],
-    lineHeight: '125%',
-    fontFamily: 'Source Sans Pro, sans-serif',
-    fontWeight: 'normal',
-    color: (props.theme as any).colors.neutral[9],
-    '-webkit-font-smoothing': 'antialiased',
-    '-moz-osx-font-smoothing': 'grayscale',
-}));
+const Text = styled.h2<TextProps>(
+    ({ level, ...props }) =>
+        themeHelper({
+            fontSize: level,
+            lineHeight: '125%',
+            fontFamily: 'Source Sans Pro, sans-serif',
+            fontWeight: 'normal',
+            color: 'neutral.9',
+        })(props),
+    sxPropHelper,
+);
```

Comment:
This syntax is freakin' me out man...

Reply:
What about it?

Reply:
You're passing in a function to a function that's returning a curried style object. That ain't weird to you?

Reply:
I'm just "old-school" I guess

Reply:
I can pull that out to a baseStyles variable or something so maybe it reads a bit better.
But basically, styled.* can take a function which it calls with the component props and then I am destructuring level so I can use it to determine the font size.
themeHelper takes an object and returns a function that takes the props because it needs that to get the theme to use it to determine the styles.

It is a little confusing... maybe we can hop on a call at some point and talk through it.